src/app/modules/persistence-service/persistence.service.ts

```diff
@@ -19,59 +24,226 @@ export abstract class AbstractPersistenceService {
   abstract getBinaryResource(path: string): Promise<Blob>;
 }
 
+export interface BackupEntry {
+  resource: string;
+  backupResource: string;
+}
+
+export interface PullResponse {
+  failure: boolean;
+  diffExists: boolean;
+  headCommit: string;
+  changedResources: Array<string>;
+  backedUpResources: Array<BackupEntry>;
+}
+
 @Injectable()
 export class PersistenceService extends AbstractPersistenceService {
 
   private serviceUrl: string;
   private listFilesUrl: string;
+  private openNonDirtyTabs: Array<string>;
+  private openDirtyTabs: Array<string>;
+  private subscriptions: Subscription[];
+
+  startSubscriptions(): void {
+    this.subscriptions = new Array();
+    this.subscriptions.push(this.messagingService.subscribe(NAVIGATION_OPEN, (id) => {
+      this.removeNonDirtyTab(id);
+      this.openNonDirtyTabs.push(id);
+    }));
+    this.subscriptions.push(this.messagingService.subscribe(NAVIGATION_CLOSE, () => {
```

Comment:
Just as a side note (I know this event is a preexisting one), `NAVIGATION_CLOSE` suggests its the opposite of `NAVIGATION_OPEN`, but that would actually be `EDITOR_CLOSE`, whereas `NAVIGATION_CLOSE` is published on logout… :confused: 
Not part of this PR, but we might want to consider renaming that event at some point.

Reply:
right, i'll open a fix for that